hw/xbox/mcpx_apu.c

```diff
@@ -802,12 +809,28 @@ static const MemoryRegionOps ep_ops = {
     .write = ep_write,
 };
 
+static void process_voice(MCPXAPUState *d,
+                          int32_t mixbins[NUM_MIXBINS][NUM_SAMPLES_PER_FRAME],
+                          uint32_t voice)
+{
+    /* FIXME: Implement */
+}
+
+/* This routine must run at 1500 Hz */
 /* TODO: this should be on a thread so it waits on the voice lock */
 static void se_frame(void *opaque)
 {
     MCPXAPUState *d = opaque;
+    int mixbin;
+    int sample;
+
     timer_mod(d->se.frame_timer, qemu_clock_get_ms(QEMU_CLOCK_VIRTUAL) + 10);
     MCPX_DPRINTF("mcpx frame ping\n");
+
+    /* Buffer for all mixbins for this frame */
+    int32_t mixbins[NUM_MIXBINS][NUM_SAMPLES_PER_FRAME] = { 0 };
```

Comment:
It would be nice if we could move the 4k of mix bin data off the stack and write directly into DSP memory here.